zrml/neo-swaps/src/lib.rs

```diff
@@ -832,9 +846,11 @@ mod pallet {
             let market = T::MarketCommons::market(&market_id)?;
             ensure!(market.status == MarketStatus::Active, Error::<T>::MarketNotActive);
             ensure!(market.scoring_rule == ScoringRule::Lmsr, Error::<T>::InvalidTradingMechanism);
-            let asset_count = spot_prices.len();
-            ensure!(asset_count as u16 == market.outcomes(), Error::<T>::IncorrectVecLen);
-            ensure!(market.outcomes() <= MAX_ASSETS, Error::<T>::AssetCountAboveMax);
+            let asset_count_u16: u16 =
+                spot_prices.len().try_into().map_err(|_| Error::<T>::NarrowingConversion)?;
+            let asset_count_u32: u32 = asset_count_u16.into();
+            ensure!(asset_count_u16 == market.outcomes(), Error::<T>::IncorrectVecLen);
+            ensure!(asset_count_u32 <= MaxAssets::get(), Error::<T>::AssetCountAboveMax);
```

Comment:
You can use shadowing here if you like. I mean this:

```rust
            let asset_count: u16 =
                spot_prices.len().try_into().map_err(|_| Error::<T>::NarrowingConversion)?;
            ensure!(asset_count_u16 == market.outcomes(), Error::<T>::IncorrectVecLen);
            let asset_count: u32 = asset_count.into();
            ensure!(asset_count_u32 <= MaxAssets::get(), Error::<T>::AssetCountAboveMax);
```

Let me know if this doesn't work.

Reply:
Ah, you use it below... mhhh... then it's fine I guess.